mtags/src/main/scala-3/scala/meta/internal/pc/PcRenameProvider.scala

```diff
@@ -0,0 +1,38 @@
+package scala.meta.internal.pc
+import scala.meta.internal.mtags.MtagsEnrichments.*
+import scala.meta.pc.OffsetParams
+
+import dotty.tools.dotc.ast.tpd.*
+import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.dotc.util.SourcePosition
+import org.eclipse.{lsp4j as l}
+final class PcRenameProvider(
+    driver: InteractiveDriver,
+    params: OffsetParams,
+    name: String,
+) extends PcCollector[l.TextEdit](driver, params):
+  val newName = name.stripBackticks.backticked
+
+  def collect(tree: Tree, pos: SourcePosition): l.TextEdit =
+    val isBackticked =
+      sourceText(pos.start) == '`' && sourceText(pos.end - 1) == '`'
+    // when the old name contains backticks, the position is incorrect
+    val isOldNameBackticked = sourceText(pos.start) != '`' &&
+      sourceText(pos.start - 1) == '`' &&
+      sourceText(pos.end) == '`'
+
+    if isBackticked then
+      l.TextEdit(pos.toLsp, "`" + newName.stripBackticks + "`")
```

Comment:
Same thing as with Scala 2 version. Why do you strip backticks and add them again? This should be covered in else. Also position should be adjusted before being given to collect.

Reply:
There are two cases: 
1. The were there as a part of the name e.g. `val 'hello-!-' = "Hello"`, then we want to remove backticks in renaming. This is `isOldNameBackticked` case. 
2. Backticks are not on declaration. We have to keep them, since we don't know if we are not inside pattern match. It worked like that in nonlocal rename. 
The positions look suspicious , but I tested it a lot and everything works fine, I'm not sure if we can simplify it

Reply:
We are stripping backticks and adding them again, since we don't know if they are there and we must add them, but we don't want to have them doubled